runtime/lua/vim/filetype/detect.lua

```diff
@@ -131,7 +134,15 @@ function M.bas(bufnr)
   vim.bo[bufnr].filetype = 'basic'
 end
 
-function M.bindzone(bufnr, default_ft) end
+function M.bindzone(bufnr, default)
+  local lines = table.concat(getlines(bufnr, 1, 4))
+  local first = '^; <<>> DiG [0-9%.]+.* <<>>'
+  if findany(lines, { first .. '%$ORIGIN', first .. '%$TTL', first .. 'IN%s+SOA' }) then
```

Comment:
The pattern in filetype.vim is ambiguous, I had to look up the file format of a bind zone file to verify:

```suggestion
  if findany(lines, { '^; <<>> DiG [0-9%.]+.* <<>>', '%$ORIGIN', '%$TTL', 'IN%s+SOA' }) then
```